src/Ssh/Ssh/Common/BastionUtils.cs

```diff
@@ -0,0 +1,318 @@
+            {
+                Name = "Developer"
+            };
+
+            var bastion = new BastionHost
+            {
+                Location = location,
+                IpConfigurations = new List<BastionHostIPConfiguration>(),
+                VirtualNetwork = virtualNetwork,
+                ScaleUnits = 2,
+                DisableCopyPaste = null,
+                EnableFileCopy = null,
+                EnableIpConnect = null,
+                EnableShareableLink = null,
+                EnableTunneling = null,
+                EnableKerberos = null,
+                EnableSessionRecording = null,
+                Sku = sku,
+                Tags = new Dictionary<string, string>()
+            };
+
+            try
+            {
+                this.BastionClient.CreateOrUpdate(resourceGroupName, bastionName, bastion);
+
+            }
+            catch (Exception ex)
+            {
+                throw new AzPSCloudException($"Error creating bastion: {ex.Message}");
+            }
+            return this.BastionClient.Get(resourceGroupName, bastionName);
+        }
+
+        protected void CheckValidBastionDeveloperLocation(string location)
+        {
+            string[] validLocations = { "centraluseuap", "eastus2euap", "westus", "northeurope", "northcentralus", "westcentralus" };
+
+            if (!Array.Exists(validLocations, element => element.Equals(location, StringComparison.OrdinalIgnoreCase)))
+            {
+                string error = ($"The Bastion Developer Sku is not currently available in the specified region." +
+                    $"Learn more here: https://learn.microsoft.com/en-us/azure/bastion/configuration-settings");
+                throw new AzPSCloudException(error);
+
+            }
+        }
+        private (string vNetId, string vNetName, string resourceGroupName) GetVNetDetailsFromSubnetId(string subnetId)
+        {
+            var parts = subnetId.Split('/');
+            int vNetIndex = Array.IndexOf(parts, "virtualNetworks");
+            int resourceGroupIndex = Array.IndexOf(parts, "resourceGroups");
+
+            if (vNetIndex == -1 || vNetIndex >= parts.Length - 1)
+            {
+                throw new ArgumentException("Invalid subnet ID", nameof(subnetId));
+            }
+            if (resourceGroupIndex == -1 || resourceGroupIndex >= parts.Length - 1)
+            {
+                throw new ArgumentException("Invalid subnet ID", nameof(subnetId));
+            }
+
+            string vNetName = parts[vNetIndex + 1];
+            string vNetId = string.Join("/", parts.Take(vNetIndex + 2));
+            string resourceGroupName = parts[resourceGroupIndex + 1];
+
+            return (vNetId, vNetName, resourceGroupName);
+        }
+
+        public static string ParseAvailableBastions(string bastionsFoundInVnet)
+        {
+            var jsonObject = JObject.Parse(bastionsFoundInVnet);
+
+            int count = (int)jsonObject["count"];
```

Comment:
What happens if `count` doesn't exist?

Reply:
The query that we are running always returns a count number, if it does not that means the resource graph query failed, but I will still place the check as well